lib/routes/recovery-key.js

```diff
@@ -0,0 +1,190 @@
+            throw errors.unverifiedSession()
+          }
+
+          return db.createRecoveryKey(uid, recoveryKeyId, recoveryData)
+        }
+
+        function emitMetrics() {
+          log.info({
+            op: 'account.recoveryKey.created',
+            uid
+          })
+
+          return request.emitMetricsEvent('recoveryKey.created', {uid})
+        }
+      }
+    },
+    {
+      method: 'GET',
+      path: '/recoveryKey/{recoveryKeyId}',
+      options: {
+        auth: {
+          strategy: 'accountResetToken'
+        },
+        validate: {
+          params: {
+            recoveryKeyId: validators.recoveryKeyId
+          }
+        }
+      },
+      handler: async function (request) {
+        log.begin('getRecoveryKey', request)
+
+        const uid = request.auth.credentials.uid
+        const ip = request.app.clientAddress
+        const recoveryKeyId = request.params.recoveryKeyId
+        let recoveryData
+
+        return customs.checkAuthenticated('getRecoveryKey', ip, uid)
+          .then(getRecoveryKey)
+          .then(() => { return {recoveryData} })
+
+        function getRecoveryKey() {
+          return db.getRecoveryKey(uid)
+            .then((res) => {
+              // `db.getRecoveryKey` doesn't require recoveryKeyId to retrieve
+              // the recovery bundle, however, we should perform a security
+              // check to ensure that the returned bundle contains the recoveryKeyId.
+              if (! butil.buffersAreEqual(res.recoveryKeyId, recoveryKeyId)) {
+                throw errors.recoveryKeyInvalid()
+              }
+              recoveryData = res.recoveryData
+            })
+        }
+      }
+    },
+    {
+      method: 'POST',
+      path: '/recoveryKey/exists',
+      options: {
+        auth: {
+          mode: 'optional',
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            email: validators.email().optional()
+          }
+        },
+        response: {
+          schema: {
+            exists: isA.boolean().required()
+          }
+        }
+      },
+      handler(request) {
+        log.begin('recoveryKeyExists', request)
+
+        const email = request.payload.email
+        let exists = false, uid
+
+        if (request.auth.credentials) {
+          uid = request.auth.credentials.uid
+        }
+
+        return Promise.resolve()
+          .then(() => {
+            if (! uid) {
+              // If not using a sessionToken, an email is required to check
+              // for a recovery key. This occurs when checking from the
+              // password reset page and allows us to redirect the user to either
+              // the regular password reset or account recovery password reset.
+              if (! email) {
+                throw errors.missingRequestParameter('email')
+              }
+
+              return customs.check(request, email, 'recoveryKeyExists')
```

Comment:
Do we need to add "recoveryKeyExists" as an action in customs-server?

Reply:
Added in https://github.com/mozilla/fxa-customs-server/pull/259